packages/react/src/components/Form/PinField.tsx

```diff
@@ -0,0 +1,157 @@
+import {ChangeEvent, KeyboardEvent, useCallback, useEffect, useRef, useState} from 'react';
+import '@compassion-gds/css/src/components/Form/pin.css';
+
+import { FieldSize, FieldState } from './Helpers';
+
+type Values = string[];
+
+interface PinProps {
+  pattern?: RegExp,
+  autoFocus?: boolean;
+  secret?: boolean;
+  state?: FieldState;
+  onComplete?: (value) => void;
+}
+
+interface PinHookProps extends PinProps {
+  values?: Values;
+  onChange?: (values: Values) => void;
+}
+
+export interface PinFieldProps extends PinProps {
+  label?: string;
+  value?: string | number;
```

Comment:
value type should be just `string` 
Right now the `number` is getting converted to string and `onComplete` is called with a `string` 

Even if originally `value` was `number` latter it gets converted to a `string`. This might be unexpected for some use cases so we can just limit the prop type to `string`